### PR TITLE
Add api documentation

### PR DESCRIPTION
**Pull request by dlai0001**
I created a docs folder and added the API documentation.  I figure this probably might grow as this library grows.  I also added an .npmignore file so non-functional code files will not get picked up by the npm install process.

btw, thanks for creating this library.  This attempt to document the API is so I can learn the code base better.  I want to in the future improve the suspicion protocol to allow other failure metrics beyond ping.  (for example error responses).  I think SWIM has great promise.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+* local.meta (optional) - Additional information to add meta data about this node.  This
+will be associated with the node identification when sending messages.
+
+## codec
+encoding to send payloads.  Default msgpack.
+* json - encodes message sent between nodes as raw json.
+* msgpack - Uses msgpack, http://msgpack.org/, encoding to reduce number of
+bytes sent by the payload.
+
+##disseminationFactor
```
Comment (mrhooray) [inline docs/api.md]: Dissemination factor can be used to fine tune the responsiveness of the cluster. Greater dissemination factor results to:
- more hosts being notified in every round of dissemination
- lower convergence time of cluster membership
- more/bigger network packets being sent

Vice versa.

File: docs/api.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+When a node encounters an unresponsive node, it will mark that node "suspicious",
+if enough nodes mark the node as suspect, then the node is dropped form the
+network.
+
+##interval
```
Comment (mrhooray) [inline docs/api.md]: Number of milliseconds between failure detections, also known as the protocol interval.  Every X milliseconds, nodes will ping a member of the SWIM network to check if its liveness with Time-Bounded Strong Completeness as described in the paper.

File: docs/api.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+Number of milliseconds between failure detection intervals.  Every X
+milliseconds, nodes will ping a member of the SWIM network to check if its peer
+is still running/responding.
+
+##joinTimeout
+Number of milliseconds before emitting a JoinTimeout error.  The node will still
+run as a base node separate from the network.
+
+##pingTimout
+Average ping response time threshold, in milliseconds, for suspicion.
```
Comment (mrhooray) [inline docs/api.md]: Number of milliseconds before sending ping-req messages to the unresponsive node.

File: docs/api.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+##pingReqTimout
+Any ping response time above this threshold, in milliseconds, will mark this
```
Comment (mrhooray) [inline docs/api.md]: Number of milliseconds elapsed from sending ping-req message before marking the unresponsive node suspicious.

File: docs/api.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+node suspicious
+
+##pingReqGroupSize
+Number of pings to use to get average response time.
```
Comment (mrhooray) [inline docs/api.md]: Number of hosts to send ping-req messages to for pinging unresponsive nodes indirectly to reduce false positives.

File: docs/api.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+## methods
+* constructor(options) - See Options Object.
+* boostrap(hostsToJoin, onBootStrapFunction) -
+- hostsToJoin - Addresses (with port) to try to connect to when joining the network.
+- onBootStrapFunction(err) - handler
+* whoami() - node identifier
+* members(hasLocal, hasFaulty) - gets a list of nodes in the network.
+hasLocal - nodes locally connected to this node.
+hasFaulty - nodes that are labeled suspect.
```
Comment (mrhooray) [inline docs/api.md]: includes nodes marked as faulty

File: docs/api.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+# Swim object
+
+## methods
+* constructor(options) - See Options Object.
+* boostrap(hostsToJoin, onBootStrapFunction) -
+- hostsToJoin - Addresses (with port) to try to connect to when joining the network.
+- onBootStrapFunction(err) - handler
+* whoami() - node identifier
+* members(hasLocal, hasFaulty) - gets a list of nodes in the network.
+hasLocal - nodes locally connected to this node.
```
Comment (mrhooray) [inline docs/api.md]: includes local/current node

File: docs/api.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+##pingReqTimout
+Any ping response time above this threshold, in milliseconds, will mark this
+node suspicious
+
+##pingReqGroupSize
+Number of pings to use to get average response time.
+
+##udp
+UDP Options
+* maxDgramSize - max size of UDP datagram before sending.
```
Comment (mrhooray) [inline docs/api.md]: Max size of UDP datagram. If bigger than what the network supports, messages might be chunked into multiple packets and discarded at receiver end.

**Comment by dlai0001** [comment]
I'll update some of the comments.  Perhaps in a separate request pull some of the more commonly used fields into the main ReadMe

**Comment by mrhooray** [comment]
thanks @dlai0001

**Pull request merged**